ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -31,6 +31,7 @@
 import com.github.ambry.store.StoreKey;
 import com.github.ambry.utils.Time;
 import com.github.ambry.utils.Utils;
+import com.google.common.collect.ImmutableList;
```

Comment:
Are we currently using google guava library? If temporarily not,  in JDK8 we can use `Collections.unmodifiableList(Arrays.asList(blobIdStr));`, or in JDK9 we can use `List.of(blobIdStr)` instead.

Reply:
You can also use Collections.singletonList() - which should be enough here.

Reply:
Changing to use Collections.singletonList().  I wasn't aware of that method.  We must be importing guava lib otherwise the build would have failed.